bentoml/_internal/configuration/containers.py

```diff
@@ -170,11 +171,11 @@ def override(self, keys: t.List[str], value: t.Any):
                 " to the required schema, key=%s, value=%s." % (keys, value)
             ) from e
 
-    def as_dict(self) -> t.Dict[str, t.Any]:
-        return self.config
+    def as_dict(self) -> providers.ConfigDictType:
+        return t.cast(providers.ConfigDictType, self.config)
 
 
-@container
+@attr.s
```

Comment:
Use `attr.define` or pyright will complain.

Reply:
attr.define broke the whole thing

Reply:
nvm

Reply:
@bojiang so you suggest not to do this?

Reply:
just dataclass here works for both pyright and runtime. actually it should be `simple_di.container` (which is just an alias of `dataclass` in that lib but sadly pyright didn't understand it)